src/libraries/System.Runtime/tests/System/Reflection/NullabilityInfoContextTests.cs

```diff
@@ -761,13 +763,18 @@ public void NullablePublicOnlyOtherTypesTest()
             info = nullabilityContext.Create(privateNullableField);
             Assert.Equal(NullabilityState.Unknown, info.ReadState);
             Assert.Equal(NullabilityState.Unknown, info.WriteState);
+
+            ConstructorInfo privateConstructor = typeof(IndexOutOfRangeException)
+                .GetConstructor(BindingFlags.NonPublic | BindingFlags.Instance, new[] { typeof(SerializationInfo), typeof(StreamingContext) })!;
+            info = nullabilityContext.Create(privateConstructor.GetParameters()[0]);
+            Assert.Equal(NullabilityState.Unknown, info.WriteState);
```

Comment:
Previously, this was returning `Nullable` because the code wasn't checking for public-only on private constructor parameters.